plugins/org.eclipse.n4js/src/org/eclipse/n4js/naming/ModuleNameComputer.xtend

```diff
@@ -68,7 +68,8 @@ class ModuleNameComputer {
 			val location = sourceContainer.location
 			if(uri.uriStartsWith(location)) {
 				var relativeURI = uri.deresolve(location.appendSegment(""))
-				if (ResourceType.xtHidesOtherExtension(uri) || (N4JSGlobals.XT_FILE_EXTENSION == uri.fileExtension.toLowerCase)) { // support Xpect tests with files having additional file extension .xt
+				 // support Xpect tests with files *.n4sj.xt and *.xt
+				if (ResourceType.xtHidesOtherExtension(uri) || (N4JSGlobals.XT_FILE_EXTENSION == uri.fileExtension.toLowerCase)) {
```

Comment:
not part of your changes, but still:
does "toLowerCase" make sense in this line? (file extensions are case sensitive, at least on some OSes)

Reply:
Also the newly added comment is misleading as this check applies to all file extensions hidden inside the .xt extension, not only to .n4js.xt

Reply:
n4sj

Reply:
ResourceType does checks in `.toLowerCase` mode so for consistency it makes sense.

Reply:
comment removed